src/replication.c

```diff
@@ -296,8 +374,12 @@ void incrementalTrimReplicationBacklog(size_t max_blocks) {
                               server.repl_backlog->histlen + 1;
 }
 
-/* Free replication buffer blocks that are referenced by this client. */
-void freeReplicaReferencedReplBuffer(client *replica) {
+/* Free replication buffer blocks that are referenced by this client.
+ * Also, removes replica from rdbchannel waiting list. */
+void freeReplicaReferences(client *replica) {
```

Comment:
i think we would better not mix up them.

and now is `server.replicas_waiting_rdbchannel` necessary, i think we can remove it, and search server.clients if needed and then check the replication state of target client if it is expected, right?

Reply:
Are you saying:
- revert this change
- call lookupClientByID() to get client, check if its state is WAIT_RDBCHANNEL.

I guess we can do it.

Reply:
yes, it is